ui/src/connection.ts

```diff
@@ -136,26 +137,50 @@ export function generateSecret(): string {
   return randomID(SECRET_LEN)
 }
 
-async function rawCommand<T>(baseUrl: string, command: unknown): Promise<T> {
-  return fetch(`${baseUrl}/command`, {
-    method: 'POST',
-    mode: 'same-origin',
-    headers: {
-      'Content-Type': 'application/json',
-    },
-    body: JSON.stringify(command),
-  }).then(res => {
-    if (res.status !== 200) {
-      throw new Error('Unexpected status code: ' + res.status);
+async function rawCommand<T>(baseUrl: string, command: unknown, attempt?: number): Promise<T> {
+  if (attempt === undefined) attempt = 1
+
+  const retry = async () => {
+    const wait = Math.min((2 ** (attempt! - 1)) * 100, 10000)
+    await sleep(wait)
+    return rawCommand<T>(baseUrl, command, attempt! + 1)
+  }
+
+  let res;
+  try {
+    res = await fetch(`${baseUrl}/command`, {
+      method: 'POST',
+      mode: 'same-origin',
+      headers: {
+        'Content-Type': 'application/json',
+      },
+      body: JSON.stringify(command),
+    })
+  } catch(e) {
+    // console.error('API command error: ', e);
+    // console.log(e)
+    // throw e;
+    return await retry()
+  }
+
+  if (res.status !== 200) {
+    const body = await res.text()
+
+    if (body === DROPPED_SESSION_BODY) {
```

Comment:
ideally we'd check the code too, it should be 400

Reply:
considering how explicit the body message is, why go more specific and create a thighter coupling?